dataset/src/main/scala/frameless/TypedEncoder.scala

```diff
@@ -327,4 +330,32 @@ object TypedEncoder {
     recordEncoder: Lazy[RecordEncoderFields[G]],
     classTag: ClassTag[F]
   ): TypedEncoder[F] = new RecordEncoder[F, G]
+
+  type UDT[A >: Null] = FramelessInternals.PublicUserDefinedType[A]
+
+  /**
+    * Encodes things using a Spark SQL's User Defined Type (UDT) if there is one defined in implicit.
+    *
+    * Example: to use Spark ML's VectorUDT implementation as a TypedEncoder, add the following in your scope:
+    * {{{
+    * import org.apache.spark.ml.linalg.{Vector, SQLDataTypes}
+    * import frameless.TypedEncoder.UDT
+    * implicit val mLVectorUDT: UDT[Vector] = SQLDataTypes.VectorType.asInstanceOf[UDT[Vector]]
```

Comment:
Yes I agree. The cast is needed here only because [org.apache.spark.ml.linalg.SQLDataTypes.VectorType](https://github.com/apache/spark/blob/master/mllib/src/main/scala/org/apache/spark/ml/linalg/SQLDataTypes.scala#L32) upcasts UserDefinedType[Vector] to DataType (as UserDefinedType is private).
To avoid a facing downcast, I could add a helper like this in TypedEncoderUtils:
```scala
def asUDTUnsafe[A >: Null](dataType: DataType): UDT[A] = dataType.asInstanceOf[UDT[A]]
```
Your case class example allows to construct an incorrect UDT as "def userDefinedType" will have to downcast DataType to UserDefinedType[A] and throws an exception for non-UDT userDefinedType, so I would prefer a more explicit "asUDTUnsafe" method. What do you think?